### DISCUSSION TITLE
Adding UNS threads

### DISCUSSION THREAD

**Discussion by d-schmidt** [adrianschlatter/threadlib#65] [Q&A]
I'm missing the US microphone thread (5/8" - 27 UNS). 
UNS uses the same 60° flank angle like other UIS threads so I naivly assume it would be rather trival to add them all or at least the microphone one?

https://www.gewinde-normen.de/en/unified-special-thread.html

**Comment by adrianschlatter** [comment]
Indeed, these threads are not included in threadlib. I'd like to invite you to add them :-) These documents could help you get started: [Design of Threadlib](https://github.com/adrianschlatter/threadlib/blob/develop/docs/DesignOfThreadlib.md), [Creating Thread Specs](https://github.com/adrianschlatter/threadlib/blob/develop/docs/CreatingThreadSpecs.md)

**Comment by d-schmidt** [comment]
Is there more to it than copy pasting the diameter/tpi into the UIS_thread.csv?
The guide doesn't offer much technical detail for a complete beginner. This is the first time I've seen an awk file? What does the existing UIS_thread.awk do? It is hard to understand what is happening without comments and single char variables.

**Comment by adrianschlatter** [comment]
I guess there _is_ a bit more to it than copy and paste. You'd have to find out the basic shape of the thread. This is probably easy: So far, a trapezoidal shape (4 points) has worked for all thread types.

Then you'd identify the tolerance ranges so that you can aim for the center (=> pen&paper work). If you are lucky, UNS threads indeed use the same formulas as UIS but you'd have to check. These formulas are then implemented in a .awk file that translates a spec table (as in your link) to threadlib specs.

Regarding awk: It translates an input table into an output table. Tutorials and documentation are available on the internet. As you can see in the design/ folder, I wrote separate awk scripts for each type of thread. I expect you'd have to write a new one for UNS, too.

**Comment by RegEM** [comment]
Thanks for your efforts Adrian.
For my needs, I used;
27-UNS-5/8	27	0.6250 in the UIS_thread.csv
Ran the appropriate awk commands.
- cat design/UIS_thread.csv | awk -f design/UIS_thread.awk >> design/THREAD_TABLE.csv
- cat design/THREAD_TABLE.csv | awk -f design/autogenerate.awk > THREAD_TABLE.scad
And then manually used it,
MY_THREAD_TABLE = [ ["27-UNS-5/8-int", [0.940741, -7.9690, 15.9298, 
                                      [[0, 0.4116], [0, -0.4116], [0.5092, -0.1176], [0.5092, 0.1176]]]]];
nut("27-UNS-5/8", turns = 16, Douter=26, higbee_arc=20, fn=120, table=MY_THREAD_TABLE);
Some problems running awk on windows, perhaps related to line feed characters or something. Not sure where my awk hails from. I just started over to fix.
It threads on my tripod ok, though perhaps a bit tight. GooglesAI says, "A 5/8 inch UNS thread has a  pitch of 0.941 millimeters" So pitch calculation looks good. Printing out a couple of eye bolts. I'll inform if I think there is a thread sizing problem.
